### PR TITLE
Scratch docs

### PR DESCRIPTION
**Pull request by jgonthier**
## Description
Update of the scratch directory documentation.

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] break up "scratch files and the ~/.psi4rc file" into separate sections so that we're not suggesting psi4rc is the best way to set your scratch
- [x] csh/bash example of setting PSI_SCRATCH
- [x] Example of a threaded job (needs approval)
- [x] Edit the link on page https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/tutorial.rst in sentence "For your convenience" to point to something you created.

## Questions
- [x] @loriab, @dgasmith : Could you have a look at the PBS job file example section and tell me if that looks convenient ?
- [x]  @loriab : I actually don't understand what you mean by your last point. Does https://github.com/psi4/psi4/blob/master/samples/tu1-h2o-energy/input.dat need an update ?

## Status
- [x] Ready to go

### CODE REVIEW

File: doc/sphinxman/source/external.rst [hunk 1]
```diff
@@ -229,6 +267,105 @@ to explicitly control the number of threads used for integral formation. Setting
...
+    #PBS -j oe 
+    #PBS -l pmem=2120mb
+    #PBS -N jobname
+    #PBS -V
+    
+The ``PBS -j oe`` option instructs PBS to write any output or error message
+from the queuing system in dedicated files. ``PBS -l pmem=2120mb`` requests 
+2120 MB of memory for each thread on the node. The total memory requested for 
+the job by PBS should generally be slightly greater than what indicated 
+with the |adc__memory| keyword in the input file.
```
Comment (loriab) [inline doc/sphinxman/source/external.rst]: why adc_memory? maybe ``:ref:`memory setting <sec:memory>` ``?

File: doc/sphinxman/source/external.rst [hunk 2]
```diff
@@ -229,6 +267,105 @@ to explicitly control the number of threads used for integral formation. Setting
...
+The next section is _very important_ as it sets the environment variables needed
+by |PSIfour|: ::
+
+    setenv PSI_SCRATCH $myscratch
+    setenv PSIPATH /path/to/external/modules:${PSIPATH}
+    setenv PSIPATH /path/to/python/modules:${PSIPATH}
+    setenv PSIDATADIR /psi/install/directory/share
+
+:envvar:`PSIDATADIR` should point to the ``share`` directory in |PSIfour| install
+directory. :envvar:`PSIPATH` is needed only if you are using external modules or
```
Comment (loriab) [inline doc/sphinxman/source/external.rst]: Add sentence here like "(Setting :envvar:`PSIDATADIR` is not necessary when running an _installed_ copy of |PSIfour|, but setting it explicitly here avoids contamination if the environment variable has already been set and points to a different version.)"

**Comment by loriab** [comment]
For the reduced PBS file, I think you'll want something like the below. For psidatadir, just ensure non-interference. need to have psipath existing before appending. and jobname must match pbs name. i've tested it on dftd3 at least.

```
#!/bin/tcsh
#PBS -j oe
#PBS -l pmem=2120mb
#PBS -N jobname
#PBS -V


setenv OMP_NUM_THREADS 4
setenv MKL_NUM_THREADS 4
cd $PBS_O_WORKDIR
setenv myscratch /scratch/user/psi4.$PBS_JOBID

foreach i (`sort $PBS_NODEFILE | uniq`)
    echo "Creating scratch directory " $myscratch " on " $i
    ssh $i rm -rf $myscratch
    ssh $i mkdir -p $myscratch
end

unsetenv PSI4DATADIR
unsetenv PSIDATADIR
setenv PSI_SCRATCH $myscratch
if ! ( $?PSIPATH ) setenv PSIPATH ""
setenv PSIPATH /path/to/external/modules:${PSIPATH}
setenv PSIPATH /path/to/python/modules:${PSIPATH}
/psi/install/directory/bin/psi4 -i jobname.in -o jobname.out

foreach i (`sort $PBS_NODEFILE | uniq`)
    echo "Removing scratch directory " $myscratch " on " $i
    ssh $i rm -rf $myscratch
end
```

File: doc/sphinxman/source/external.rst (hunk 1 above)
Comment (jgonthier) [inline doc/sphinxman/source/external.rst]: I wanted to refer to the memory keyword for Psi4, and the only one I could find was the adc__memory. Is the "memory" we specify in the input file not a keyword ?

**Comment by jgonthier** [comment]
Fine by me ! I'll make the final modifications today. However, the PBS job name does not have to correspond to the name of the input file (if that is what you meant), I just ran H2O_HO2.in with the name "Patate_kiwi".

Also, what about the last point in psi4#109 ? I still don't understand what you meant.

File: doc/sphinxman/source/external.rst (hunk 1 above)
Comment (loriab) [inline doc/sphinxman/source/external.rst]: Correct, "memory" is not a keyword but a command. It lacks "set" but gains units. What you're actually calling is http://psicode.org/psi4manual/master/autodoc_driver.html#p4util.util.set_memory .

**Comment by loriab** [comment]
Huh, never mind the jobname bit then. I was changing various things so I must have been mistaken on that point.

Ah, in tutorial.rst, there's a sentence below. That ``sec:install_III_7`` link is to a long-deceased file. Please replace that link with one to the "7." on the wiki page or anywhere else you think best covers the subject.

```
"For your convenience, the above example can be found in :srcsample:`tu1-h2o-energy`. You can run it if you wish. Once |PSIfour| is in your path (see the :ref:`User Configuration <sec:install_III_7>` section"
```

**Comment by jgonthier** [comment]
Everything should be done now.

**Pull request merged**